### PR TITLE
[ADD] feat(PDF-attestation): New user actions to generate PDF attesta…

### PR DESCRIPTION
**Pull request by Pourbaix**
Adding new users actions to be able to download a PDF attestation from the Frontend.

+ New button in 'MyDspace' section, for items that have been archived, in order to download the attestation.
+ New action option on the detail view of an archived item that allows an admin or the submitter to download the attestation.

### CODE REVIEW

File: src/app/core/data/pdf-attestation-download.service.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { NotificationsService } from 'src/app/shared/notifications/notifications.service';
+
+@Injectable()
+export class PdfAttestationDownloadService {
+
+    // Injecting httpClient into this class
+    constructor(private httpClient: HttpClient,
+        protected notificationsService: NotificationsService,
+        protected translationService: TranslateService
+        ){}
```
Comment (zannkukai) [inline src/app/core/data/pdf-attestation-download.service.ts]: Why `protected` and not just `private` ?
Maybe a little nit more indentation could be great : 
```typescript
constructor(
    private httpClient: HttpClient,
    private notificationService: NotificationService,
    private translationService: TranslateService
){ }
```

File: src/app/core/data/pdf-attestation-download.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { Injectable } from '@angular/core';
+import { environment } from 'src/environments/environment';
+import { Observable, catchError, of } from 'rxjs';
+import { TranslateService } from '@ngx-translate/core';
+import { NotificationsService } from 'src/app/shared/notifications/notifications.service';
+
+@Injectable()
+export class PdfAttestationDownloadService {
+
+    // Injecting httpClient into this class
```
Comment (zannkukai) [inline src/app/core/data/pdf-attestation-download.service.ts]: Check and complete docstring

File: src/app/core/data/pdf-attestation-download.service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    constructor(private httpClient: HttpClient,
+        protected notificationsService: NotificationsService,
+        protected translationService: TranslateService
+        ){}
+
+    /**
+     * Generic method to make a HTTP GET call
+     */
+    public get(url: string) {
```
Comment (zannkukai) [inline src/app/core/data/pdf-attestation-download.service.ts]: Is this method need to be `public` ? `private` seems sufficent

File: src/app/core/data/pdf-attestation-download.service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                );
+                return of();
+            }))
+            .subscribe((response) => {
+                const attestationBlob = response;
+                const data = window.URL.createObjectURL(attestationBlob);
+                const link = document.createElement('a');
+                link.href = data;
+                link.download = 'attestation';
+                link.click();
```
Comment (zannkukai) [inline src/app/core/data/pdf-attestation-download.service.ts]: Maybe better ? to be tested
```typescript
import FileSaver from 'file-saver';
...
var blob = new Blob([response], {type: "application/pdf"});
FileSaver.saveAs(blob, "filename");
```
https://stackoverflow.com/a/59161634/5595377

File: src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.scss
Comment (zannkukai) [inline src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.scss:1]: remove this file as not used

File: src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     * @param {DSpaceObjectType} injectedContextMenuObjectType
+     * @param {AuthorizationDataService} authorizationService
+     * @param notificationService
+     */
+    constructor(
+        @Inject('contextMenuObjectProvider') protected injectedContextMenuObject: DSpaceObject,
+        @Inject('contextMenuObjectTypeProvider') protected injectedContextMenuObjectType: DSpaceObjectType,
+        protected authorizationService: AuthorizationDataService,
+        private notificationService: NotificationsService,
+        protected pdfAttestationDownloadService: PdfAttestationDownloadService,
```
Comment (zannkukai) [inline src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.ts]: at least this service could be `private` ? Check others

File: src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        protected translationService: TranslateService
+    ) {
+        super(injectedContextMenuObject, injectedContextMenuObjectType, ContextMenuEntryType.PdfAttestation);
+    }
+
+    ngOnInit() {
+        this.notificationService.claimedProfile.subscribe(() => {
+            this.isAuthorized$ = this.authorizationService.isAuthorized(FeatureID.CanDownloadPDFAttestation, this.contextMenuObject.self, undefined, false);
+        });
+    }
```
Comment (zannkukai) [inline src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.ts]: You need to unsubscribe in `ngOnDestroy`
```typescript
export class PDFAttestationMenuComponent extends ContextMenuEntryComponent implements OnInit, OnDestroy {

private subscriptions = new Subscription();
...
ngOnInit() {
    this.subscriptions.add(
        this.notificationService.claimedProfile.subscribe(() => { ... })
    );
}

ngOnDestroy() {
    this.subscriptions.unsubscribe();
}
```

File: src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+    genAttestationDownloadURL(){
```
Comment (zannkukai) [inline src/app/shared/context-menu/pdf-attestation/pdf-attestation-menu.component.ts]: please use complete function name --> `generateAttestationDownloadURL`

File: src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    public constructor(
+        protected pdfAttestationDownloadService: PdfAttestationDownloadService,
+        protected notificationsService: NotificationsService,
+        protected translationService: TranslateService,
+        protected authorizationService: AuthorizationDataService
+    ) {
+    }
+
+    ngOnInit(): void {
+        this.notificationsService.claimedProfile.subscribe(() => {
```
Comment (zannkukai) [inline src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.ts]: implement `ngOnDestroy` and unsubscribe this event into it

File: src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            this.isAuthorized$ = this.authorizationService.isAuthorized(FeatureID.CanDownloadPDFAttestation, this.object.self, undefined, false);
+        });
+        this.initDSOAttestationRoute(this.object.id);
+    }
+
+    initDSOAttestationRoute(itemID: string) {
```
Comment (zannkukai) [inline src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.ts]: private ?

File: src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        this.objectId = itemID;
+    }
+
+    genAttestationDownloadURL() {
```
Comment (zannkukai) [inline src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.ts]: fullname

File: src/assets/i18n/en.json5 [hunk 11]
```diff
@@ -6594,6 +6600,8 @@
 
   "subscriptions.table.empty.message": "You do not have any subscriptions at this time. To subscribe to email updates for a Community or Collection, use the subscription button on the object's page.",
 
+  "thesis.page.titleprefix": "Thesis:",
```
Comment (zannkukai) [inline src/assets/i18n/en.json5:6603]: remove trailing space will make github happy :-)

File: src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.scss
Comment (zannkukai) [inline src/app/shared/mydspace-actions/pdf-attestation-action/pdf-attestation-action.component.scss:1]: remove as not used

File: src/app/core/data/pdf-attestation-download.service.ts [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+                this.notificationsService.error(
+                    this.translationService.get('submission.workflow.generic.attestation.error.title'),
+                    this.translationService.get('submission.workflow.generic.attestation.error.content')
+                );
+                return of();
+            }))
+            .subscribe((response) => {
+                const attestationBlob = response;
+                saveAs(attestationBlob, 'attestation.pdf');
+        });
```
Comment (zannkukai) [inline src/app/core/data/pdf-attestation-download.service.ts]: COSMETIC : could be shorter `.subscribe((response) => saveAs(response, 'attestation.pdf'))`

**Review by zannkukai — APPROVED** [review]

File: src/app/core/data/pdf-attestation-download.service.ts (hunk 12 above)
Comment (Pourbaix) [reply src/app/core/data/pdf-attestation-download.service.ts]: I'm a dumpster

**Pull request merged**